app/models/manageiq/providers/ibm_power_hmc/infra_manager/event_target_parser.rb

```diff
@@ -31,11 +31,15 @@ def parse
         $ibm_power_hmc_log.info("#{self.class}##{__method__} LPAR uuid #{uuid}")
         target_collection.add_target(:association => :vms, :manager_ref => {:ems_ref => uuid})
       when "VirtualSwitch"
-        $ibm_power_hmc_log.info("#{self.class}##{__method__} VirtualSwitch uuid #{uuid}")
-        target_collection.add_target(:association => :hosts, :manager_ref => {:ems_ref => elems[-3]})
+        if elems.length >= 4 && elems[-4] == "ManagedSystem"
```

Comment:
This is looking complex enough that we might want to split out parsing of the URI elements into its own method that returns a hash with a key describing what each element is.  A casual reader will have no idea what `elems[-4]` is but `elems[:type]` and `elems[:subtype]` or something like that is much more readable.

Reply:
Ok